From: XSC Project <xsc@example.com>
Subject: [PATCH] Export poll/select internal functions for XSC

Export do_sys_poll and kern_select for use by XSC ring-based
syscall module.

Signed-off-by: XSC Project <xsc@example.com>
---
 fs/select.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs/select.c b/fs/select.c
index 0ee55af..f8c9e3a 100644
--- a/fs/select.c
+++ b/fs/select.c
@@ -1006,6 +1006,7 @@ static int do_sys_poll(struct pollfd __user *ufds, unsigned int nfds,
 	}
 	return err;
 }
+EXPORT_SYMBOL(do_sys_poll);

 static long do_restart_poll(struct restart_block *restart_block)
 {
@@ -733,6 +734,7 @@ static int kern_select(int n, fd_set __user *inp, fd_set __user *outp,
 	}
 	return ret;
 }
+EXPORT_SYMBOL(kern_select);

 SYSCALL_DEFINE5(select, int, n, fd_set __user *, inp, fd_set __user *, outp,
 		fd_set __user *, exp, struct __kernel_old_timeval __user *, tvp)
--
2.34.1
